pkg/apis/machineconfiguration.openshift.io/v1/helpers.go

```diff
@@ -20,13 +20,15 @@ func NewMachineConfigPoolCondition(condType MachineConfigPoolConditionType, stat
 
 // GetMachineConfigPoolCondition returns the condition with the provided type.
 func GetMachineConfigPoolCondition(status MachineConfigPoolStatus, condType MachineConfigPoolConditionType) *MachineConfigPoolCondition {
+	// in case of sync errors, return the last condition that matches, not the first
```

Comment:
This makes sense but I was under the impression that Conditions were unique, as in if I had a degraded condition, a new one would simply overwrite it instead of adding to the list, is that not the case? (I'm probably wrong, I haven't looked at this in awhile, but I can't recall what determines this so curious if you know)

Reply:
I was mostly leaving this here for redundancy since the other logic `could` have a race at some point and I feel like it is just easier to always be sure you use the "last" instance of a state. I'd be fine with removing it if you want to :)

Reply:
Makes sense! I think it's fine to leave it as is, thanks!